### ISSUE TITLE
Upgrade Helm chart to 3.0 Existing S3 Storage not read

### ISSUE THREAD

**Issue by asaf400** [grafana/loki#7685]
This may or may not be an issue with Loki & the Helm Charts, but here goes:

When attempting to upgrade an existing Release of Chart v2.12.2 - Loki v2.5.0,
which is running as a single-binary deployment but **with** aws s3 storage with boltdb-shipper

and upgrade to Chart v3.3.4 - Loki v2.6.1 (Latest)
with the simple-scalable configuration,

I am noticing that when Grafana queries any logs from upgraded deployment, 
it would only show the logs that were collected from the moment the new deployment started, and all the log history is 'gone'
and when reverting the release back to Chart v2.12.2 the all the existing logs show up again..

I have tried following these referenced:
https://grafana.com/docs/loki/next/installation/helm/upgrade-from-2.x/
https://github.com/grafana/loki/blob/main/production/helm/loki/values.yaml - latest
https://github.com/grafana/helm-charts/blob/loki-2.12.2/charts/loki/values.yaml - existing

Now of course the values.yaml are vastly different between the charts,
while it was quite difficult to migrate the configurations, more then the upgrade guide led to believe,
I was able to make it work, and so I'll paste both the values.yaml files, and the rendered /etc/loki/loki.yaml from within the pod..

v2.12.2 provided values.yaml:
https://pastebin.com/EKHTuvc8

v2.12.2 Rendered:
https://pastebin.com/93RuShuG

v3.3.4 provided values.yaml:
https://pastebin.com/BFS3GQKn

v3.3.4 Rendered:
https://pastebin.com/wMgQXbx0

Attached are grafana explore screenshots that show a gap in which Chart v2 was used,
So loki with the V3 chart only sees logs created with V3,
and V2 only sees V2 (Versions of the chart)

![Loki Chart V2](https://user-images.githubusercontent.com/14169006/201613477-dbddf9d3-ec08-4c11-80c4-f85d01223062.png)

![Loki Chart v3](https://user-images.githubusercontent.com/14169006/201613473-96c370c1-0564-4ddd-9340-0e9db77ccbea.png)

In summary, something in the way the V3 chart configures loki makes it so that the s3 storage is weird.. 
or something is missing in my configuration but I have no idea what..

Edit: The logs are of an app that created 1 line every 5 seconds.

**Comment by tbondarchuk** [comment]
@asaf400 I've just upgraded my loki setup and got it to work with both old and new logs with following values:

```yaml
loki:
  auth_enabled: false
  storage:
    bucketNames:
      chunks: BUCKET
      ruler: BUCKET
      admin: BUCKET
    s3:
      region: REGION
  schemaConfig:
    configs:
    - from: 2021-01-01
      store: boltdb-shipper
      object_store: s3
      schema: v11
      index:
        prefix: index_
        period: 24h
    - from: 2022-11-21
      store: boltdb-shipper
      object_store: s3
      schema: v12
      index:
        prefix: loki_index_
        period: 24h
```

old values:
```yaml
config:
  schema_config:
    configs:
    - from: 2021-01-01
      store: boltdb-shipper
      object_store: aws
      schema: v11
      index:
        prefix: index_
        period: 24h
  storage_config:
    boltdb_shipper:
      active_index_directory: /data/loki/boltdb-shipper-active
      cache_location: /data/loki/boltdb-shipper-cache
      cache_ttl: 24h
      shared_store: aws
    aws:
      s3: s3://REGION/BUCKET
  compactor:
    working_directory: /data/loki/boltdb-shipper-compactor
    shared_store: aws
```

I have a suspicion that storage name must be consistent everywhere and new chart seems to use name 's3' rather then 'aws' so I've just went with almost all default values from new chart, just added schemaConfig to point to old logs with v11 schema, based on [this comment](https://github.com/grafana/loki/issues/7267#issuecomment-1264472870)

I did lost day of logs so on next cluster will try to switch to v12 and new index prefix on old chart and upgrade next day.

Looking at your values I guess I'd try to remove `storage_config.aws` section since storage is already configured in `storage.s3` and change `schemaConfig` to `object_store: s3`.
